### PR TITLE
 License Plate Detection And Recognition Using OpenCv And Pytesseract

### LINKED ISSUE #5255: (Python) License Plate Detection And Recognition Using Opencv And Pytesseract
### Topic Suggestion
(Python) License Plate Detection & Recognition Using Opencv And Pytesseract

### Writing sample(s): 
[Sample](https://www.section.io/engineering-education/create-a-screen-recorder-using-python-and-pycharm/#:~:text=%20Creating%20a%20Screen%20Recorder%20Using%20Python%20and,Datetime%3A%20We%20will%20need%20this%20package...%20More%20)
#### Proposed title of article
License plate detection and recognition using opencv and pytesseract

#### Proposed article introduction
License plate detection and recognition software are used in traffic management, car identification, surveillance, etc. You may have seen them in action in your favorite tech-based movie. You probably have always been curious about how they accomplish this.

Python helps us accomplish this using the following libraries:

- opencv
- imutils
- pytesseract

In this tutorial, we will learn how to use the above libraries together with Python for license plate detection and recognition. 

#### Key takeaways 

- The processes a software passes to detect and recognize a license plate.
- What are the above-mentioned libraries and how they are used in license plate detection and recognition.
- How to create a  program that detects and recognizes license plates using opencv and pytesseract.

#### Article quality
This article will combine both license plate detection and recognition. It will also have a well-explained code, not just a copied and pasted one.

#### References
N/A

### PR DESCRIPTION
**Pull request by kirurisimon**
### Article checklist

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

`This closes #5255

### CODE REVIEW

**Comment by lalith1403** [comment]
plagiarism check done ✔️

File: content/articles/license-plate-detection-and-recognition-using-opencv-and-pytesseract/index.md [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+images:
+
+ - url: /engineering-education/license-plate-detection-and-recognition-using-opencv-and-pytesseract/hero.jpg
+   alt: License Plate image
+---
+License plate detection is identifying the part of the car that is predicted to be the number plate. Recognition is identifying the values that make up the license plate.
+<!--more-->
+License plate detection and recognition is the technology that uses computer vision to detect and recognize a license plate from an input image of a car. License plate detection is identifying the part of the car that is predicted to be the number plate. Recognition is identifying the values that make up the license plate.
+
+This technology applies in many areas. On roads, it is used to identify the cars that are breaking the traffic rules. On security, it is used to capture the license plates of the vehicles getting into and out of certain premises. In parking lots, it is used to capture the license plates of the cars being packed. The list of its applications goes on and on.
```
Comment (collins-a) [inline content/articles/license-plate-detection-and-recognition-using-opencv-and-pytesseract/index.md:21]: Pay closer attention to grammar in this paragraph. _On security_ does not sound right in the second line. _Packed_ is incorrect in the second last line.

File: content/articles/license-plate-detection-and-recognition-using-opencv-and-pytesseract/index.md [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+- [Prerequisites](#prerequisites)
+- [Processes a software undergoes to detect and recognize a license plate](#processes-a-software-undergoes-to-detect-and-recognize-a-license-plate)
+- [Creating a license plate detection and recognition program](#creating-a-license-plate-detection-and-recognition-program)
+- [Results](#results)
+- [Conclusion](#conclusion)
+
+### Prerequisites
+To follow through this tutorial, a person needs to:
+- Be familiar with the **python** programming language
+- Have **pycharm** installed on their computer. You can download **pycharm community edition** from here. [Pycharm Community Edition ](https://www.jetbrains.com/pycharm/download/)
```
Comment (collins-a) [inline content/articles/license-plate-detection-and-recognition-using-opencv-and-pytesseract/index.md:38]: It's much more fitting having the link on the word **here** as opposed to adding a phrase after it.

File: content/articles/license-plate-detection-and-recognition-using-opencv-and-pytesseract/index.md [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+- `cv2.destroyAllWindows()`: We are closing all the open windows.
+### Results
+When the program has run successfully, its output is as follows:
+
+![Results](/engineering-education/license-plate-detection-and-recognition-using-opencv-and-pytesseract/results.gif)
+
+### Conclusion
+You now have all the skills required to create a program that detects and recognizes license plates. Capture more images of vehicles and input them into the program then sit back and watch as the program does its magic.
+
+! Happy coding
```
Comment (collins-a) [inline content/articles/license-plate-detection-and-recognition-using-opencv-and-pytesseract/index.md:232]: Send the exclamation mark to the end of the phrase. Also, consider always adding a few references/ further reading links for the reader to get more content on the topic.

**Review by collins-a — CHANGES_REQUESTED** [review]
@kirurisimon good tutorial on plate detection and recognition. A few very minor changes and we shall move on to the final review.

Additionally, where applicable in future tutorials, consider using notebooks such as Google Colab (and share the link at the end of the tutorial) to allow readers to follow along and tweak the code for themselves without having to install an IDE.

Ping me when done.

**Comment by kirurisimon** [comment]
@collins-a 
Finished working on the changes.

**Review by collins-a — APPROVED** [review]

**Comment by ahmadmardeni1** [comment]
Great tutorial @kirurisimon 
Thank you for the review. @collins-a

**Pull request merged**